airgun/entities/host_new.py

```diff
@@ -32,6 +38,163 @@ def get_details(self, entity_name, widget_names=None):
+
+        if not add_to_all_collections:
+            view.searchbar.fill("name = " + host_collection_name, enter_timeout=5)
+            view.wait_displayed()
+            self.browser.plugin.ensure_page_safe()
+            if view.host_collection_table.row_count == 0:
+                raise ValueError(f'{host_collection_name} not found in host collections!')
+            time.sleep(3)
+            view.host_collection_table[0][0].widget.click()
+        else:
+            view.select_all.click()
+        view.add.click()
+
+    def remove_host_from_host_collection(
+        self, entity_name, host_collection_name=None, remove_from_all_collections=False
+    ):
+        """
+        Function that removes host from host collection
+
+        Args:
+            entity_name: Name of the host
+            host_collection_name: Name of the host collection we want to remove host from
+            remove_from_all_collections: If True, host will be removed from all host collections
+
+        Raises:
+            ValueError: If specific hostColName is set to be removed & remove all is set to True.
+            ValueError: If no parameters are passed.
+            ValueError: If host is not assigned to any host collection.
+            ValueError: If given host col name we want to remove is not assigned to host.
+        """
+
+        if (host_collection_name is not None) and remove_from_all_collections:
+            raise ValueError(
+                'Either host_collection_name or remove_from_all_collections must be provided!'
+            )
+        if (host_collection_name is None) and (remove_from_all_collections is False):
+            raise ValueError(
+                'host_collection_name None and remove_from_all_collections False '
+                'is invalid combination!'
+            )
+
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        # Handle the case where there are no host collections assigned
+        view.overview.click()
+        if view.overview.host_collections.no_host_collections.is_displayed:
+            raise ValueError('No host collections assigned to host, thus nothing to remove!')
+
+        view.overview.host_collections.kebab_menu.item_select('Remove host from collections')
+        view = ManageHostCollectionModal(self.browser)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+
+        if not remove_from_all_collections:
+            view.searchbar.fill("name = " + host_collection_name, enter_timeout=5)
+            view.wait_displayed()
+            self.browser.plugin.ensure_page_safe()
+            if not view.host_collection_table.is_displayed:
+                raise ValueError(
+                    f"{host_collection_name} not assigned to host, thus can't remove it!"
+                )
+            time.sleep(3)
```

Comment:
Is this stable enough? If not, you can `wait_for()`

Reply:
I have tested it several times and it is working.

Reply:
I have added this as an extra precaution because it was sometimes reloading search result two times in a row.